File: csharp/FlyweightPattern/Course.cs

using System;
namespace FlyweightPattern
{
    public class Course : AbstractCourse
    {
        /// <summary>
        /// 课程名称
        /// </summary>
        private string name;

        /// <summary>
        /// 课时
        /// </summary>
        private int periods;

        public Course(string name, int periods)
        {
            this.name = name;
            this.periods = periods;
        }

        public override void Print(Student student)
        {
            Console.WriteLine(student.Name + ":\t" + this.name + "(" + this.periods + "课时)");
        }
    }
}

File: c++/FlyweightPattern/Course.cpp
#include "stdafx.h"
#include "Course.h"


Course::Course(string name, int periods)
{
	this->name = name;
	this->periods = periods;
}


void Course::Print(Student student)
{
	cout<<student.name << ":\t" << this->name << "(" << this->periods << "¿ÎÊ±)"<<endl;;
}
